test_plans/multi_ptr_test_plan.asciidoc

```diff
@@ -0,0 +1,431 @@
+
+* `access::decorated::yes`
+* `access::decorated::no`
+
+Check call and const correctness prefetch(size_t numElements)
+
+==== Arithmetic operators
+
+For the following values as Space parameter:
+
+* `access::address_space:global_space`
+* `access::address_space:local_space`
+* `access::address_space:private_space`
+* `access::address_space:generic_space`
+
+For the following values as DecorateAddress parameter:
+
+* `access::decorated::yes`
+* `access::decorated::no`
+
+Constract multi_ptr with accessor to 5th element of array of 10 consecutive values starting with 1 of testing type
+with same template parameters and check result of operators:
+
+* `multi_ptr& operator++(multi_ptr& mp)` refers to `6`
+* `multi_ptr operator++(multi_ptr& mp)` refers to `6`
+* `multi_ptr& operator--(multi_ptr& mp)` refers to `4`
+* `multi_ptr operator--(multi_ptr& mp, int)` refers to `4`
+* `multi_ptr& operator+=(multi_ptr& lhs, difference_type r)` with `r == 3` refers to `8`
+* `multi_ptr& operator-=(multi_ptr& lhs, difference_type r)` with `r == 2` refers to `3`
+* `multi_ptr operator+(const multi_ptr& lhs, difference_type r)` with `r == 4` refers to `9`
+* `multi_ptr operator-(const multi_ptr& lhs, difference_type r)`  with `r == 1` refers to `4`
+* `reference operator*(const multi_ptr& lhs)` equals `5`
+
+Constract 2 multi_ptr's m_ptr1 and m_ptr2 with accessors to 1th and 2nd elements of array of testing type
+with same template parameters and check result of operators:
+
+* `bool operator==(const multi_ptr& lhs, const multi_ptr& rhs)`
+
+`true` for `m_ptr1` and `m_ptr1`
+`false` for `m_ptr1` and `m_ptr2`
+
+* `bool operator!=(const multi_ptr& lhs, const multi_ptr& rhs)`
+
+`false` for `m_ptr1` and `m_ptr1`
+`true` for `m_ptr1` and `m_ptr2`
+
+
+* `bool operator<(const multi_ptr& lhs, const multi_ptr& rhs)`
+
+`true` for `m_ptr1` and `m_ptr2`
+`false` for `m_ptr2` and `m_ptr1`
+
+* `bool operator>(const multi_ptr& lhs, const multi_ptr& rhs)`
+
+`false` for `m_ptr1` and `m_ptr2`
+`true` for `m_ptr2` and `m_ptr1`
+
+* `bool operator<=(const multi_ptr& lhs, const multi_ptr& rhs)`
+
+`true` for `m_ptr1` and `m_ptr1`
+`true` for `m_ptr1` and `m_ptr2`
+`false` for `m_ptr2` and `m_ptr1`
+
+* `bool operator>=(const multi_ptr& lhs, const multi_ptr& rhs)`
+
+`true` for `m_ptr1` and `m_ptr1`
+`true` for `m_ptr2` and `m_ptr1`
+`false` for `m_ptr1` and `m_ptr2`
+
+Constract multi_ptr with accessor or with nullptr_t
+with same template parameters and check result of operators:
+
+* `bool operator==(const multi_ptr& lhs, std::nullptr_t)`
+
+`true` for m_ptr constracted from nullptr_t
+`false` for m_ptr constracted from accessor
+
+* `bool operator!=(const multi_ptr& lhs, std::nullptr_t)`
+
+`false` for m_ptr constracted from nullptr_t
+`true` for m_ptr constracted from accessor
+
+* `bool operator<(const multi_ptr& lhs, std::nullptr_t)`
+
+std::less<multi_ptr<value_type, Space, access::decorated::yes>::pointer>()(multi_ptr.get(), nullptr)
```

Comment:
Interesting... Do you think we should check all the these comparisons?
Looking at for example https://stackoverflow.com/questions/65185742/can-you-compare-nullptr-to-other-pointers-for-order-is-it-always-smaller
it is unclear.
Do we say anything in the spec about this?

Reply:
SYCL 2020 spec only says about these operators 'Comparison operator OP for multi_ptr class with a std::nullptr_t'
And since we need to check the result somehow I thought we can use std::less etc, similarly how it's described for `unique_ptr` and `std::nullptr_t` https://en.cppreference.com/w/cpp/memory/unique_ptr/operator_cmp